kits/daisy/src/robot/hexapod.cpp

```diff
@@ -0,0 +1,610 @@
+      && hex_errors.has_valid_initial_feedback;
+    hex_errors.first_out_of_range_leg = getFirstOutOfRange(positions_);
+    hex_errors.m_stop_pressed = getMStopPressed(fbk);
+  }
+
+  // TODO: generalize!
+  legs_.emplace_back(new Leg(30.0 * M_PI / 180.0, 0.2375, getLegFeedback(0), params, real_legs_.count(0)>0, 0, Leg::LegConfiguration::Left));
+  legs_.emplace_back(new Leg(-30.0 * M_PI / 180.0, 0.2375, getLegFeedback(1), params, real_legs_.count(1)>0, 1, Leg::LegConfiguration::Right));
+  legs_.emplace_back(new Leg(90.0 * M_PI / 180.0, 0.1875, getLegFeedback(2), params, real_legs_.count(2)>0, 2, Leg::LegConfiguration::Left));
+  legs_.emplace_back(new Leg(-90.0 * M_PI / 180.0, 0.1875, getLegFeedback(3), params, real_legs_.count(3)>0, 3, Leg::LegConfiguration::Right));
+  legs_.emplace_back(new Leg(150.0 * M_PI / 180.0, 0.2375, getLegFeedback(4), params, real_legs_.count(4)>0, 4, Leg::LegConfiguration::Left));
+  legs_.emplace_back(new Leg(-150.0 * M_PI / 180.0, 0.2375, getLegFeedback(5), params, real_legs_.count(5)>0, 5, Leg::LegConfiguration::Right));
+
+  // Initialize step information
+  last_step_legs_.insert(0);
+  last_step_legs_.insert(3);
+  last_step_legs_.insert(4);
+
+  last_fbk = std::chrono::steady_clock::now();
+  // Start a background feedback handler
+  if (group_)
+  {
+    group->addFeedbackHandler([this] (const GroupFeedback& fbk)
+    {
+      Eigen::Vector3d avg_accel;
+      avg_accel << 0, 0, 0; // unnecessary?
+      Eigen::Vector3d avg_gyro;
+      avg_gyro << 0, 0, 0; // unnecessary?
+
+      std::lock_guard<std::mutex> guard(fbk_lock_);
+      last_fbk = std::chrono::steady_clock::now();
+      assert(fbk.size() == Leg::getNumJoints() * real_legs_.size());
+
+      // Copy data into an array
+      copyIntoPositions(positions_, &fbk, real_legs_);
+
+      // Get averaged body-frame IMU data for each leg
+      // TODO: For each, CHECK THIS IS VALID/HAS FEEDBACK!
+      int num_leg_joints = Leg::getNumJoints();
+      int num_legs_used = std::max((int)real_legs_.size(), 1);
+      int leg_count = 0;
+      for (int i = 0; i < 6 && leg_count < num_legs_used; ++i)
+      {
+        if (real_legs_.count(i) == 0)
+        {
+          continue;
+        }
+        leg_count++;
+        // Get from base of each leg
+        int num_prev_legs = std::count_if(real_legs_.begin(), real_legs_.end(), [i](int other_leg) { return other_leg < i; });
+        auto accel = fbk[num_prev_legs * num_leg_joints].imu().accelerometer().get(); 
+        auto gyro = fbk[num_prev_legs * num_leg_joints].imu().gyro().get(); 
+        Eigen::Vector3d mod_accel;
+        mod_accel << accel.getX(), accel.getY(), accel.getZ();
+        Eigen::Vector3d mod_gyro;
+        mod_gyro << gyro.getX(), gyro.getY(), gyro.getZ();
+        // Transform
+        Eigen::Matrix4d trans = legs_[i]->getKinematics().getBaseFrame();
+        avg_accel += trans.topLeftCorner(3,3) * mod_accel;
```

Comment:
`trans.topLeftCorner<3, 3>()` gives better compile time opportunities for codegen

Reply:
fixed.